Online/Entity/OnlinePhysicalObject.cs

```diff
@@ -556,10 +556,15 @@ public void WeaponHitSomething(RealizedWeaponState statewhenhit, OnlineCollision
                 statewhenhit.ReadTo(this);
                 SharedPhysics.CollisionResult? result = null;
                 hit.BuildCollisionResult(out result);
-                if (result.HasValue) {
+                if (result.HasValue)
+                {
+                    if (result.Value.obj != null && result.Value.obj is Creature c)
+                    {
+                        c.killTag = (this.apo.realizedObject as Weapon).thrownBy.abstractCreature;
```

Comment:
Could the kill tag be timing out?
```suggestion
                        c.SetKillTag((this.apo.realizedObject as Weapon).thrownBy.abstractCreature);
```

Reply:
need to test this. I think it's from the fact that each time it failed to register it also notified in weapon state that it would not update state because it was transferring to new owner so I need to wrap that transfer around the weapon mode instead of the whole state because we are ALSO using thrownBy in the weapon state so it seems like it should trigger the killtag if that's set correctly versus being in the RPC.


But if the state has a risk of transferring then it seems like the thrownBy in the state is only useful part of the time.